kb/python-hello-world-activity.md

```diff
@@ -0,0 +1,302 @@
+import logging
+from dataclasses import dataclass
+from datetime import timedelta
+```
+
+The `asyncio` library is used to make asynchronous calls, which is important to most applications.
+
+The `logging` library is used to log data.
+
+The `dataclass` and `datetime` libraries are used in this application.
+
+## Application code
+
+The following section covers the parameters that will be sent to the Activity and Workflow.
+
+### Data classes
+
+While we could use multiple parameters in the Activity, Temporal strongly encourages using a single data class instead, which can have fields added to it in a backwards-compatible way.
+
+```python
+@dataclass
+class ComposeGreetingInput:
+    greeting: str
+    name: str
+```
+
+## Temporal application code
+
+The following sections describe aspects of application using Temporal specific features.
+
+## Activity Definition
+
+We mark Activity Definitions with a decorator. The following Activity logs and preforms a string concatenation.
+
+```python
+@activity.defn
+async def compose_greeting(input: ComposeGreetingInput) -> str:
+    activity.logger.info("Running activity with parameter %s" % input)
+    return f"{input.greeting}, {input.name}!"
+```
+
+## Workflow Definition
+
+Workflow Definition are classes and are marked with a decorator, `@workflow.defn`. The Workflow must be defined on the class given to the Worker and the Workflow Type, or Workflow name, can be customized with a `name` parameter inside the decorator.
+
+The method invoked for the Workflow is decorated with `@workflow.run`. Exactly one method name must have this decorator and must be defined on an `async def` method. The method’s arguments are the Workflow’s arguments, which in this example takes: the Activity name, an object or data class, and a Start to Close Timeout.
+
+```python
+@workflow.defn
+class GreetingWorkflow:
+    @workflow.run
+    async def run(self, name: str) -> str:
+        workflow.logger.info("Running workflow with parameter %s" % name)
+        return await workflow.execute_activity(
+            compose_greeting,
+            ComposeGreetingInput("Hello", name),
+            start_to_close_timeout=timedelta(seconds=10),
+        )
+```
+
+## Invoke the Client
+
+### Start Client
+
+Clients don’t have explicit close statements. The Client pass two arguments in this example, a target host and a namespace. Since we are running this locally, we use `[localhost:7233](http://localhost:7233)` which is the location that Temporalite is running on.
+
+```python
+async def main():
+    client = await Client.connect("localhost:7233", namespace="default")
+```
+
+## Invoke the Worker
+
+### Start the Worker
+
+The Worker Program defines the `client`, `task_queue`, `workflows`, and `activities`.
+
+A Worker Entity listens and polls on a single Task Queue. A Worker Entity contains both a Workflow Worker and an Activity Worker so that it may make progress of either a Workflow Execution or an Activity Execution.
```

Comment:
```suggestion
A Worker Entity listens and polls on a single Task Queue.
A Worker Entity contains both a Workflow Worker and an Activity Worker so that it can make progress of either a Workflow Execution or an Activity Execution.
```